mozilla_version/fenix.py

```diff
@@ -0,0 +1,77 @@
+"""Defines characteristics of a Fenix version at Mozilla."""
+
+import attr
+import re
+
+from mozilla_version.version import BaseVersion
+from mozilla_version.parser import strictly_positive_int_or_none
+
+
+@attr.s(frozen=True, cmp=False, hash=True)
+class FenixVersion(BaseVersion):
+    """Class that validates and handles Fenix version numbers."""
+
+    _VALID_ENOUGH_VERSION_PATTERN = re.compile(r"""
+        ^(?P<major_number>\d+)
+        \.(?P<minor_number>\d+)
+        (\.(?P<patch_number>\d+))?
+        (
+            -beta\.(?P<beta_number>\d+)
+            |-rc\.(?P<release_candidate_number>\d+)
+        )?$""", re.VERBOSE)
+
+    _OPTIONAL_NUMBERS = BaseVersion._OPTIONAL_NUMBERS + (
+        'beta_number', 'release_candidate_number'
+    )
+
+    _ALL_NUMBERS = BaseVersion._ALL_NUMBERS + _OPTIONAL_NUMBERS
+
+    beta_number = attr.ib(type=int, converter=strictly_positive_int_or_none, default=None)
+    release_candidate_number = attr.ib(
+        type=int, converter=strictly_positive_int_or_none, default=None
+    )
+
+    @classmethod
+    def parse(cls, version_string):
+        """Construct an object representing a valid Fenix version number."""
+        return super(FenixVersion, cls).parse(version_string)
```

Comment:
Nit: I don't think we need to re-define the parse method, since we just delegate the call to the parents' method.